server/web_ui/admin/src/components/admin_objectgraph.rs

```diff
@@ -0,0 +1,294 @@
+            } else {
+                None
+            }
+        }).collect::<Vec<_>>();
+
+        // Constructing graph source
+        let mut sb = String::new();
+        sb.push_str("digraph {\n  rankdir=\"RL\"\n");
+        for (spn, _, members) in members_of {
+            members.iter()
+                .filter(|member| typed_entries.iter().any(|(spn, _, _)| spn == *member))
+                .for_each(|member| {
+                    sb.push_str(format!(r#"  "{spn}" -> "{member}"{}"#, "\n").as_str());
+                });
+        }
+
+        for (spn, uuid, obj_type) in typed_entries {
+            let (color, shape, route) = match obj_type {
+                ObjectType::Group => ("#b86367", "box", AdminRoute::ViewGroup { uuid }),
+                ObjectType::BuiltinGroup => ("#8bc1d6", "component", AdminRoute::ViewGroup { uuid }),
+                ObjectType::ServiceAccount => ("#77c98d", "parallelogram", AdminRoute::ViewServiceAccount { uuid }),
+                ObjectType::Person => ("#af8bd6", "ellipse", AdminRoute::ViewPerson { uuid }),
+            };
+            let url = route.to_path();
+            sb.push_str(format!(r#"  "{spn}" [color = "{color}", shape = {shape}, URL = "{url}"]{}"#, "\n").as_str());
+        }
+        sb.push_str("}");
+        init_graphviz(&sb.as_str());
+
+        let on_checkbox_click = {
+            let scope = ctx.link().clone();
+            move |_: Event| {
+                let coll = document().get_elements_by_class_name("obj-graph-filter-cb");
+                let mut filters = vec![];
+
+                for i in 0..*&coll.length() {
+                    let option = coll.get_with_index(i)
+                        .expect("couldnt get elem between 0 and selection length ???");
+                    let input_el = option.unchecked_into::<HtmlInputElement>();
+                    let checked = input_el.checked();
+
+                    if checked {
+                        let value = input_el.id();
+                        let obj_type = ObjectType::try_from(value).expect("Option attribute —value— is not a valid ObjectType");
+                        filters.push(obj_type);
+                    }
+                }
+                scope.send_message(Msg::NewFilters { filters });
+            }
+        };
+
+        let view_graph_source = {
+            move |_: MouseEvent| {
+                open_blank(sb.as_str());
+            }
+        };
+
+        html! {
+            <>
+            <div class={CSS_PAGE_HEADER}>
+            <h2>{ "ObjectGraph view" }</h2>
+            </div>
+            if entries.is_empty() {
+                <div>
+                  <h5>{ "No graph objects for the applied filters." }</h5>
+                </div>
+            } else {
+                <div class="column">
+                    <div class="hstack gap-3">
+                    {
+                        all::<ObjectType>().map(|ot| {
+                            let str = format!("{:?}", ot);
```

Comment:
```suggestion
                            let str = format!("{:?}", ot);
```

Can we please do an `impl Display` for `ObjectType` and have:

- Group -> "Group" 
- BuiltinGroup -> "Built In Group" 
- etc

(it feels like "Group" should be "Local Group" or something to distinguish from Builtin Group, but I'll ponder that and it's fine for now)